test/python/circuit/test_controlled_gate.py

```diff
@@ -781,6 +781,14 @@ def test_open_controlled_gate_raises(self):
         with self.assertRaises(CircuitError):
             base_gate.control(num_ctrl_qubits, ctrl_state='201')
 
+    def test_open_controlled_x_gate_eq(self):
+        """Test open control x gate equality."""
+        circuit1 = QuantumCircuit(2)
+        circuit1.append(XGate().control(1, ctrl_state='1'), [0, 1])
+        circuit2 = QuantumCircuit(2)
+        circuit2.append(XGate().control(1, ctrl_state='1'), [0, 1])
```

Comment:
I think these should be `ctrl_state='0'` to reproduce the original bug.

Reply:
Sigh, yeah I made a mistake when I was trying to get to the bottom of it but forgot to update the test. It turns out this isn't actually a fix the infinite recursion, it's still there. I'll push up a revert in a sec and I've reopened the original issue.

Reply:
Reverted in: https://github.com/Qiskit/qiskit-terra/pull/4516